pkg/eth/tx_listener.go

```diff
@@ -68,7 +72,7 @@ func (t *TxListener) StopSubscription(receipt *jobs.Subscription) error {
 
 // PollForTxCompletion will continuously poll until a transaction is complete
 func (t *TxListener) PollForTxCompletion(txID string, updates chan<- string) {
-
+	go t.waitForTimeout()
```

Comment:
This probably shouldn't be part of the struct, but relative to this invocation of `PollForTxCompletion`. What if the listener has more than 1 tx it is listening to? Could you just make this a go func with `var timeoutPassed` at the top of the function?